### ISSUE TITLE
Scaling panopes down casues large response times

### ISSUE THREAD

**Issue by camallen** [zooniverse/Panoptes#687]
![hmm_blocking_io](https://cloud.githubusercontent.com/assets/295329/6721872/e94168bc-cdc9-11e4-9272-05a11be4ccda.jpg)

It might be linked to cellect going offline and the api tries sending http get requests for subjects and waiting on the responses. Or it might not be blocking and something else is happening. I dont think api nodes shutting down would cause this spike in ruby time either. @edpaget thoughts?

2 cellect nodes went down
1st one:
2015 March 18 23:31:59 UTC
2015 March 18 23:33:01 UTC
2nd one:
2015 March 18 23:26:28 UTC
2015 March 18 23:27:30 UTC

3 API nodes went down
1:
2015 March 18 23:36:11 UTC
2015 March 18 23:37:15 UTC
2:
2015 March 18 23:35:40 UTC
2015 March 18 23:36:48 UTC
3:
2015 March 18 23:25:39 UTC
2015 March 18 23:26:42 UTC

**Comment by edpaget** [comment]
It looks like its the reload users in Cellect that caused the issue. The workflows_controller#show method spiked at the same time mostly due to the load_user cellect method.

**Comment by camallen** [comment]
Should we have a timeout on the cellect http request so we spin waiting for the request to timeout?

**Comment by edpaget** [comment]
Maybe we should load users in a sidekiq worker?

**Comment by camallen** [comment]
Great idea for the workflow async load that totally doesn't need to be in the request response cycle. Not sure about get subjects though. At the least it won't stop the site loading to the home page.

**Comment by edpaget** [comment]
Hmm that might be trickier than I thought unless we want to store the cellect host for users somewhere on the server (redis maybe?.) We need to be able to set the cellect host into the user session.

**Comment by camallen** [comment]
Can't we just pass this down from the session into the worker via the same method we use for cellect client [here](https://github.com/zooniverse/Panoptes/blob/93628d669c585064e0197b2ec55148719cb69f1f/lib/cellect_client.rb#L77)?

**Comment by edpaget** [comment]
We need to set it back into the session though. We pass the whole session object to the CellectClient, so it can set new hosts into the session. We don't have access to a user session object in sidekiq since it's not part of the request/response cycle.

**Comment by camallen** [comment]
Yeah - that's only if the cellect server has disappeared though. We can get the CellectClient to choose the host and set it in the session before serialising the Cellect host ip into the worker, so any previous deaths should get caught and reset in that request cycle.

Chances of the cellect host dying between the request coming in and being reprocessed by sidekiq are slim but it could happen. This operation is an optimization on the user seen set load time so it's nice to get it in before they ask for a subject but not completly necessary for cellect to work. Cellect may take a hit on setting up the data structures and loading data if it comes with a get request as well. I think the worst i saw of this in my benchmarking of 85% seen set for 500K was ~10seconds to load seen set and select the unseen set.
